route-snapper/src/lib.rs

```diff
@@ -0,0 +1,329 @@
+use geom::{Circle, Distance, FindClosest, LonLat, PolyLine, Pt2D};
+use map_model::{IntersectionID, Map, PathConstraints, RoadID};
+use wasm_bindgen::prelude::*;
+
+const INTERSECTON_RADIUS: Distance = Distance::const_meters(10.0);
+
+#[wasm_bindgen]
+pub struct JsRouteSnapper {
+    map: Map,
+    snap_to_intersections: FindClosest<IntersectionID>,
+    route: Route,
+    mode: Mode,
+}
+
+#[derive(Clone, PartialEq)]
+struct Route {
+    waypoints: Vec<IntersectionID>,
+    full_path: Vec<IntersectionID>,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+enum Mode {
+    Neutral,
+    Hovering(IntersectionID),
+    Dragging { idx: usize, at: IntersectionID },
+}
+
+#[wasm_bindgen]
+impl JsRouteSnapper {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Result<JsRouteSnapper, JsValue> {
+        // Panics shouldn't happen, but if they do, console.log them.
+        console_error_panic_hook::set_once();
+
+        let map: Map = abstutil::from_binary(include_bytes!("../bristol.bin"))
```

Comment:
Here's the hack! https://github.com/acteng/atip/tree/route-snapper-fetch-gunzip uses a dynamic fetch, but serde is blowing up somehow